docs/source/torch.compiler_dynamic_shapes.md

```diff
@@ -1,129 +1,474 @@
+(user_annotations)=
+### User Annotations
 
-## Overall architecture
+Several tools allow users to explicitly mark specific inputs
+by name or code as dynamic. This is useful for avoiding initial compilations that
+would eventually become dynamic with the previous tools. It is also used to mark
+elements that do not automatically get marked as dynamic, such as neural network
+module parameters, and so on. These tools include:
 
-Symbolic shapes workflow:
+(dynamic_sources_allow_list)=
+#### Dynamic Allow List (DYNAMIC_SOURCES)
 
-1. When we start compiling a frame in Dynamo, we allocate a ShapeEnv (attached to FakeTensorMode) which keeps track of symbolic shapes state.
-2. We allocate symbolic sizes for tensors on entry (what is static or dynamic is a policy decision, with some knobs).
-3. We propagate the symbolic sizes through operators, maintaining both (1) FX IR so that we can faithfully export symbolic compute, and (2) Sympy expressions representing the size vars, so we can reason about them.
-4. When we condition on symbolic sizes, either in Dynamo tracing or in Inductor optimization, we add guards based on the conditional. These can be induced from both Python and C++.
-5. These guards can induce further simplifications on symbolic variables. For example, if you assert `s0 == 4`, we can now replace all occurrences of `s0` with `4`.
-6. When we're done tracing and optimizing, we install all of these guards with the compiled code; the compiled code is only reusable if all the guards evaluate true.
+Use the evnironmental variable `TORCH_COMPILE_DYNAMIC_SOURCES` to pass a configuration
+list of source names to be marked as dynamic. For example:
+`TORCH_COMPILE_DYNAMIC_SOURCES=L[‘x’],L[‘y’]`
+It's easiest to find these dynamic source names using the PGO artifact in `tlparse`.
+See {ref}`identifying-dynamic-elements-marked-by-pgo` for more details. You can
+copy and paste the dynamic source names from the PGO artifact. This method works
+for integers and tensor sizes and has the highest precedence over all other flags
+that force static shapes. It will not throw an error if what is marked dynamic
+gets specialized or if the provided input does not exist.
 
-Important files:
+#### mark_dynamic(tensor, size)
 
-- C++ SymInt API: `c10/core/SymInt.h`, `SymFloat.h`, `SymBool.h`
-- Python SymInt API: `torch/__init__.py` (look for `SymInt/SymFloat/SymBool`)
-- C++ plumbing: `c10/core/SymNodeImpl.h`, `torch/csrc/utils/python_symnode.h`, `torch/csrc/jit/python/init.cpp`
-- Python infrastructure: `torch/fx/experimental/symbolic_shapes.py`
-- Other important files: `torch/_subclasses/fake_tensor.py`, `torch/_meta_registrations.py`, decomps, PrimTorch refs
+The `mark_dynamic` function marks a tensor dimension as dynamic and will fail if it
```

Comment:
These should also have a copy-pastable example. eg. you should have a reference to the full path like the current docs do `torch._dynamo.mark_dynamic(tensor, dim, min=min, max=max)`. Or at the very least give the fully qualified import statement for mark_dynamic.

Reply:
added  {func}`torch._dynamo.maybe_mark_dynamic` 
added `mark_dynamic(tensor, dim, min, max)` to the title.

Reply:
added